Firestore/core/src/firebase/firestore/nanopb/writer.h

```diff
@@ -74,7 +54,44 @@ class Writer {
   explicit Writer(const pb_ostream_t& stream) : stream_(stream) {
   }
 
-  pb_ostream_t stream_;
+  pb_ostream_t stream_{};
+};
+
+/**
+ * Creates a Writer that writes into a vector of bytes.
+ *
+ * This is roughly equivalent to the nanopb function pb_ostream_from_buffer(),
+ * except that ByteStringWriter manages the buffer.
+ */
+class ByteStringWriter : public Writer {
+ public:
+  ByteStringWriter();
+
+  nanopb::ByteString ToByteString() const;
+
+  /**
+   * Returns the vector backing this ByteStringWriter.
+   */
+  std::vector<uint8_t> ToVector();
```

Comment:
The implementation of this function surprised me; I expected it to return a copy or maybe a const reference.

In bytes_string.h, you've named a similar method 'release', and added a docstring indicating ownership transfer. Consider something like that here too.

Reply:
Yeah, I wasn't super-happy with this. Fixed.